platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/NamedVariantIdentifier.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts;
+
+import org.gradle.api.artifacts.component.ComponentIdentifier;
+import org.gradle.internal.component.model.VariantIdentifier;
+
+/**
+ * Identifier for a variant of a component, identified by its name.
```

Comment:
IIUC, this is used in places that are not "real variants"? e.g., as part of the root node in a graph.

Would a name like `AdhocVariantIdentifier` or `OpaqueVariantIdentifier` be a better name?

On my first pass through this, I thought `NamedVariantIdentifier` was the "normal" one and `NamedModuleVariantIdentifier` was the unusual one.